### PR TITLE
User Notifications

### PR DESCRIPTION
**Pull request by shagedorn**
Still needs testing on older devices and in Xcode 7.

### CODE REVIEW

**Comment by shagedorn** [comment]
Fixes https://github.com/iosphere/ISHPermissionKit/issues/46

File: ISHPermissionKit/ISHPermissionCategory.h [hunk 1]
```diff
@@ -207,6 +207,21 @@ typedef NS_ENUM(NSUInteger, ISHPermissionCategory) {
...
     ISHPermissionCategorySpeechRecognition NS_ENUM_AVAILABLE_IOS(10_0) = 10000,
+
+    /**
+     *  Modern way to request permission to local and remote notifications.
+     *
+     *  If your app supports older deployment targets, you can request this
+     *  category before requesting ISHPermissionCategoryNotificationLocal/
+     *  ISHPermissionCategoryNotificationRemote. In that case, users on
+     *  modern systems will only see one permission request if they make
+     *  a decision, but may be prompted twice if they choose "Later".
```
Comment (shagedorn) [inline ISHPermissionKit/ISHPermissionCategory.h]: We should probably change the wording to discourage this...

Comment (felixLam) [inline ISHPermissionKit/ISHPermissionCategory.h]: Wording should probably reflect that you should make sure to only ask for either of those depending on the user's system.

**Comment by felixLam** [comment]
:shipit:

**Comment by shagedorn** [comment]
Btw, it is not entirely clear whether this permission is required for silent pushes. Both of this is in the official documentation:

> If the local or remote notifications of your app or app extension interact with the user in any way, you must call this method to request authorization for those interactions

vs.

> Always call this method before scheduling any local notifications and before registering with the Apple Push Notification Service.

https://developer.apple.com/reference/usernotifications/unusernotificationcenter/1649527-requestauthorization

Will file a radar for clarification.

**Comment by shagedorn** [comment]
rdar://28027664

**Pull request merged**